### PR TITLE
Closes #2688: `operator_test.py` conversion for new test framework

### LINKED ISSUE #2688: `operator_test.py` conversion to new framework
Convert `operator_test.py` to use new test framework

### PR DESCRIPTION
**Pull request by pierce314159**
This PR (closes #2688) converts `operator_test.py` to new test framework

### CODE REVIEW

File: PROTO_tests/tests/operator_test.py [hunk 1]
```diff
@@ -0,0 +1,774 @@
...
+    elif dtype == "bool":
+        return np.random.randint(0, 1, size=size, dtype=dtype)
+    return None
+
+
+class TestOperator:
+    def test_numpy_equivalency(self, size=100, verbose=pytest.verbose):
+        # ignore numpy warnings like divide by 0
+        np.seterr(all="ignore")
+        global pdarrays
```
Comment (Ethan-DeBandi99) [inline PROTO_tests/tests/operator_test.py:28]: Maybe I am missing something, but I don't think we need the global definitions here. Was there a reason we made these global definitions?

File: PROTO_tests/tests/operator_test.py [hunk 2]
```diff
@@ -0,0 +1,774 @@
...
+            for expression, err in results["neither_implement"]:
+                print(expression)
+        print(f'# ops implemented by numpy but not arkouda: {len(results["numpy_minus_arkouda"])}')
+        if verbose:
+            for expression, err, flag in results["numpy_minus_arkouda"]:
+                print(expression)
+        print(f'# ops implemented by arkouda but not numpy: {len(results["arkouda_minus_numpy"])}')
+        if verbose:
+            for expression, res, flag in results["arkouda_minus_numpy"]:
+                print(expression, " -> ", res)
```
Comment (Ethan-DeBandi99) [inline PROTO_tests/tests/operator_test.py:149]: These could all be wrapped in a single `if verbose` it looks like

File: PROTO_tests/tests/operator_test.py [hunk 3]
```diff
@@ -0,0 +1,774 @@
...
+        print(f"  Matching results:         {matches} / {nboth}")
+        print(f"  Arkouda execution errors: {len(execerrors)} / {nboth}")
+        if verbose:
+            print("\n".join(map(": ".join, execerrors)))
+        print(f"  Dtype mismatches:         {len(dtypeerrors)} / {nboth}")
+        if verbose:
+            print("\n".join(map(": ".join, dtypeerrors)))
+        print(f"  Value mismatches:         {len(valueerrors)} / {nboth}")
+        if verbose:
+            print("\n".join(map(": ".join, valueerrors)))
```
Comment (Ethan-DeBandi99) [inline PROTO_tests/tests/operator_test.py:173]: Again, we could use a single `if verbose` block here.

File: PROTO_tests/tests/operator_test.py [hunk 4]
```diff
@@ -0,0 +1,774 @@
...
+        assert ak_invert.to_list() == np_invert.tolist()
+
+    def test_uint_bool_binops(self):
+        # Test fix for issue #1932
+        # Adding support to binopvv to correctly handle uint and bool types
+        ak_uint = ak.arange(10, dtype=ak.uint64)
+        ak_bool = ak_uint % 2 == 0
+        assert (ak_uint + ak_bool).to_list() == (ak.arange(10) + ak_bool).to_list()
+
+    def test_float_uint_binops(self):
```
Comment (Ethan-DeBandi99) [inline PROTO_tests/tests/operator_test.py:235]: I wish we could parameterize the operator here, but I am not sure we can without a lot of headache.

File: PROTO_tests/tests/operator_test.py [hunk 5]
```diff
@@ -0,0 +1,774 @@
...
+        )
+        bi_arr = ak.array(
+            [0, 1, 2, 3, 4, 2**64 - 5, 2**64 - 4, 2**64 - 3, 2**64 - 2, 2**64 - 1],
+            dtype=ak.bigint,
+            max_bits=64,
+        )
+        np_arr = np.array(
+            [0, 1, 2, 3, 4, 2**64 - 5, 2**64 - 4, 2**64 - 3, 2**64 - 2, 2**64 - 1],
+            dtype=np.uint,
+        )
```
Comment (Ethan-DeBandi99) [inline PROTO_tests/tests/operator_test.py:454]: Would it make sense to add a helper function?

File: PROTO_tests/tests/operator_test.py [hunk 6]
```diff
@@ -0,0 +1,774 @@
...
```
Comment (Ethan-DeBandi99) [inline PROTO_tests/tests/operator_test.py:498]: Again, maybe a helper function just to reduce the number of places required for potential updates in the future?

**Review by Ethan-DeBandi99 — COMMENTED** [review]
Few things to look at, but overall looks good.

File: PROTO_tests/tests/operator_test.py (hunk 1 above)
Comment (pierce314159) [reply PROTO_tests/tests/operator_test.py:28]: to be honest, this was just leftover from how it used to be in `run_test` where that lived outside the class. I think you're right that it doesn't need to be that way anymore. I'll update this

File: PROTO_tests/tests/operator_test.py (hunk 3 above)
Comment (pierce314159) [reply PROTO_tests/tests/operator_test.py:173]: I believe this the same deal, we only wanna print the mismatches with the appropriate header

File: PROTO_tests/tests/operator_test.py [hunk 7]
```diff
@@ -0,0 +1,774 @@
...
+NUMERIC_TYPES = ["int64", "float64", "bool", "uint64"]
+
+
+def make_np_arrays(size, dtype):
+    if dtype == "int64":
+        return np.random.randint(-(2**32), 2**32, size=size, dtype=dtype)
+    elif dtype == "uint64":
+        return np.random.randint(-(2**32), 2**32, size=size).astype(dtype)
+    elif dtype == "float64":
+        return np.random.uniform(-(2**32), 2**32, size=size)
```
[the remark this answers is not in the document]
Comment (pierce314159) [reply PROTO_tests/tests/operator_test.py:18]: yah I think we should add an issue for this. Right now we only do float edge case testing on `//` and `%`

File: PROTO_tests/tests/operator_test.py (hunk 4 above)
Comment (pierce314159) [reply PROTO_tests/tests/operator_test.py:235]: One idea I had was something like the below, but i feel like that ends up being less readable and harder to tell if we're testing the right thing

```python
for lfn, rfn in zip([`__add__`, `__sub__`, `__mul__`, ...], [`__radd__`, `__rsub__`, `__rmul__`, ...]):
    ak_ans = getattr(ak.pdarrayclass, fn)(ak_uint, akf)
    np_ans = getattr(np.whereever, fn)(np_uint, npf)
    assert np.allclose(ak_ans.to_ndarray(), np_ans, equal_nan=True)
```

File: PROTO_tests/tests/operator_test.py (hunk 5 above)
Comment (pierce314159) [reply PROTO_tests/tests/operator_test.py:454]: sure! I can make that change

File: PROTO_tests/tests/operator_test.py (hunk 2 above)
Comment (Ethan-DeBandi99) [reply PROTO_tests/tests/operator_test.py:149]: There are 3 separate loops, all inside a separate `if verbose`. Why can we not have a single `if` and then the loops under?

File: PROTO_tests/tests/operator_test.py (hunk 3 above)
Comment (Ethan-DeBandi99) [reply PROTO_tests/tests/operator_test.py:173]: Same response as above.

File: PROTO_tests/tests/operator_test.py (hunk 1 above)
Comment (pierce314159) [reply PROTO_tests/tests/operator_test.py:28]: Upon trying to delete the globals, I found that `do_op` requires these to be globals in order for the `eval` to work properly
```python
def do_op(lt, rt, ls, rs, isarkouda, oper):
    evalstr = ""
    if ls:
        evalstr += f'scalars["{lt}"]'
    else:
        evalstr += f'{("ndarrays", "pdarrays")[isarkouda]}["{lt}"]'
    evalstr += f" {oper} "
    if rs:
        evalstr += f'scalars["{rt}"]'
    else:
        evalstr += f'{("ndarrays", "pdarrays")[isarkouda]}["{rt}"]'
    res = eval(evalstr)
    return res
```

Comment (Ethan-DeBandi99) [reply PROTO_tests/tests/operator_test.py:28]: Ah ok. Sounds good. As long as there is a reason for them.

**Review by Ethan-DeBandi99 — APPROVED** [review]
